keps/sig-node/4381-dra-structured-parameters/README.md

```diff
@@ -1210,230 +955,202 @@ server-side-apply (SSA) simpler. Patching individual list elements is not
 needed and there is a single owner.
 
 ```go
-// ResourceSlice provides information about available
-// resources on individual nodes.
+// One or more slices represent a pool of devices managed by a given driver.
+// How many slices the driver uses to publish that pool is driver-specific.
+// Each device in a given pool must have a unique name.
+//
+// The slice in which a device gets published may change over time. The unique identifier
+// for a device is the tuple `<driver name>/<pool name>/<device name>`. Driver name
+// and device name don't contain slashes, so it is okay to concatenate them
+// like this in a string with a slash as separator. The pool name itself may contain
+// additional slashes.
+//
+// Whenever a driver needs to update a pool, it bumps the pool generation number
+// and updates all slices with that new number and any new device definitions. A consumer
+// must only use device definitions from slices with the highest generation number
+// and ignore all others.
+//
+// If necessary, a consumer can check the number of total devices in a pool (included
+// in each slice) to determine whether its view of a pool is complete.
+//
+// For devices that are not local to a node, the node name is not set. Instead,
+// a pool name is chosen by the driver as necessary. It can be left blank if the
+// device names are unique in the entire cluster. The unique identifier then
+// has the same format as before with `<driver name>[/<pool name>]/<device name>`.
 type ResourceSlice struct {
     metav1.TypeMeta
     // Standard object metadata
+    // +optional
```

Comment:
How can this be optional?

Reply:
This is common practice - perhaps because in some scenarios (embedding?) the field may be empty?

https://github.com/kubernetes/kubernetes/blob/541cfa2b87c341e6789fed8a6f22277185ab151e/staging/src/k8s.io/api/core/v1/types.go#L4758-L4761